### DISCUSSION TITLE
Adding a new algorithm based on the paper: "Solving the Many to Many assignment"

### DISCUSSION THREAD

**Discussion by MightyArty** [networkx/networkx#7450] [Algorithms]
Hello, my name is Tom and I'm a 3-year Computer Science student.
As a part of my last course at the University, I picked an interesting paper named: **"Solving the Many to Many Assignment Problem by Improving the Kuhn–Munkres Algorithm with Backtracking"**, by the authors: 

1. Haibin Zhu
2. Dongning Liu
3. Siqin Zhang
4. Yu Zhu
5. Luyao Teng
6. Shaohua Teng

My question is in which exact part of the library this algorithm could fit the most.

**Comment by Schefflera-Arboricola** [comment]
probably in a different module I think, or maybe in the [matching module](https://github.com/networkx/networkx/blob/0e7bbe5aa8879c32470343a6596c82b4d5fb523b/networkx/algorithms/matching.py) or the [bipartite.matching](https://github.com/networkx/networkx/blob/0e7bbe5aa8879c32470343a6596c82b4d5fb523b/networkx/algorithms/bipartite/matching.py), if it is bipartite specific. But, I think you would probably know the best.

- for reference: https://www.sciencedirect.com/science/article/pii/S0304397516000037
- related: https://github.com/networkx/networkx/discussions/5969
- maybe relevant(since matching and assignment problem are very similar): https://github.com/networkx/networkx/issues/5823

Thank you :)
